posthog/src/main/java/com/posthog/java/PostHog.java

```diff
@@ -34,8 +36,17 @@ public Builder host(String host) {
             return this;
         }
 
+        public Builder logger(PostHogLogger logger) {
+            this.logger = logger;
+            return this;
+        }
+
         public PostHog build() {
-            this.sender = new HttpSender.Builder(apiKey).host(host).build();
+            if (this.logger == null) {
+                this.logger = new DefaultPostHogLogger();
+            }
```

Comment:
```suggestion
            if (logger == null) {
                logger = new DefaultPostHogLogger();
            }
```

there are no 2 logger vars in this context to use `this`

Reply:
ooh I didn't know you could omit the `this` like that, nice

made me realise I could just init the logger in the base class anyway